pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -458,6 +458,41 @@ type EndpointIdentity struct {
 	LabelsSHA256 string `json:"labelsSHA256,omitempty"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CiliumIdentity is an identity managed by Cilium
+type CiliumIdentity struct {
+	// +k8s:openapi-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// SecurityLabels is the source-of-truth set of labels for this identity. The
+	// labels placed in metav1.ObjectMeta are there for convenience when
+	// searching and manipulating the CRD.
+	SecurityLabels map[string]string `json:"securityLabels"`
```

Comment:
Also, wayyy out of scope. Did you consider this to be `pkg/labels/labels.Labels` instead of a `map[string]string`? We can discuss offline.

Reply:
Hmm, I hadn't really. I guess it should work fine and might pretty-print better if we ever add that to labels.Labels. I'll try it out.

Reply:
ah, it does work but it looks a bit funky in json:
```
       {
            "apiVersion": "cilium.io/v2",
            "kind": "CiliumIdentity",
            "metadata": {
                "creationTimestamp": "2019-07-16T14:31:37Z",
                "generation": 1,
                "labels": {
                    "class": "deathstar",
                    "io.cilium.k8s.policy.cluster": "default",
                    "io.cilium.k8s.policy.serviceaccount": "default",
                    "io.kubernetes.pod.namespace": "default",
                    "org": "empire"
                },
                "name": "4508",
                "resourceVersion": "38127",
                "selfLink": "/apis/cilium.io/v2/ciliumidentities/4508",
                "uid": "d09ed5d6-35b4-4283-840e-a6832ffdf371"
            },
            "security-labels": {
                "class": {
                    "key": "class",
                    "source": "k8s",
                    "value": "deathstar"
                },
                "io.cilium.k8s.policy.cluster": {
                    "key": "io.cilium.k8s.policy.cluster",
                    "source": "k8s",
                    "value": "default"
                },
                "io.cilium.k8s.policy.serviceaccount": {
                    "key": "io.cilium.k8s.policy.serviceaccount",
                    "source": "k8s",
                    "value": "default"
                },
                "io.kubernetes.pod.namespace": {
                    "key": "io.kubernetes.pod.namespace",
                    "source": "k8s",
                    "value": "default"
                },
                "org": {
                    "key": "org",
                    "source": "k8s",
                    "value": "empire"
                }
            }
        }
```